src/github.com/getlantern/flashlight/client/handler.go

```diff
@@ -170,18 +170,11 @@ func respondOK(writer io.Writer, req *http.Request) error {
 	return resp.Write(writer)
 }
 
-func respondBadGateway(w io.Writer, msg string) {
+func respondBadGateway(resp http.ResponseWriter, msg string) {
 	log.Debugf("Responding BadGateway: %v", msg)
-	resp := &http.Response{
-		StatusCode: http.StatusBadGateway,
-		ProtoMajor: 1,
-		ProtoMinor: 1,
-	}
-	err := resp.Write(w)
-	if err == nil {
-		if _, err = w.Write([]byte(msg)); err != nil {
-			log.Debugf("Error writing error to io.Writer: %s", err)
-		}
+	resp.WriteHeader(http.StatusBadGateway)
```

Comment:
It's Intended to return HTTP 502, but it actually return 200 OK with the response as content. Fixed it.